iree/hal/vulkan/native_descriptor_set_layout.cc

```diff
@@ -56,10 +56,12 @@ static iree_status_t iree_hal_vulkan_create_descriptor_set_layout(
   }
 
   VkDescriptorSetLayoutBinding* native_bindings = NULL;
-  IREE_RETURN_IF_ERROR(iree_allocator_malloc(
-      logical_device->host_allocator(),
-      binding_count * sizeof(VkDescriptorSetLayoutBinding),
-      (void**)&native_bindings));
+  if (binding_count > 0) {
```

Comment:
nit: can move the for loop into this if to make it clearer that it's a combined bit of logic (vs implicit the loop will not execute because binding_count == 0)